core/services/pipeline/task_params.go

```diff
@@ -71,6 +71,51 @@ func (s *StringParam) UnmarshalPipelineParam(val interface{}) error {
 	return errors.Wrapf(ErrBadInput, "expected string, got %T", val)
 }
 
+func (s *StringParam) String() string {
+	if s == nil {
+		return ""
+	}
+	return string(*s)
+}
+
+type StringSliceParam []string
+
+func (s *StringSliceParam) UnmarshalPipelineParam(val interface{}) error {
+	var ssp StringSliceParam
+	switch v := val.(type) {
+	case nil:
+		ssp = nil
+	case string:
+		return s.UnmarshalPipelineParam([]byte(v))
+
+	case []byte:
+		var theSlice []string
+		err := json.Unmarshal(v, &theSlice)
+		if err != nil {
+			return errors.Wrap(ErrBadInput, err.Error())
+		}
+		*s = StringSliceParam(theSlice)
+		return nil
+	case []string:
+		ssp = v
+	case []interface{}:
+		return s.UnmarshalPipelineParam(SliceParam(v))
+	case SliceParam:
+		for _, x := range v {
+			var s StringParam
+			err := s.UnmarshalPipelineParam(x)
+			if err != nil {
+				return err
+			}
+			ssp = append(ssp, s.String())
+		}
+	default:
+		return errors.Wrapf(ErrBadInput, "expected slice, got %T", val)
```

Comment:
Is it important to say _string_ slice?
```suggestion
		return errors.Wrapf(ErrBadInput, "expected string slice, got %T", val)
```